modules/using-the-build-time-network-policy-generator.adoc

```diff
@@ -0,0 +1,76 @@
+// Module included in the following assemblies:
+//
+// * cli/generating-build-time-network-policies.adoc
+:_content-type: PROCEDURE
+[id="using-the-build-time-network-policy-generator_{context}"]
+= Using the build-time network policy generator
+
+.Prerequisites
+
+. The build-time network policy generator recursively scans the directory you specify when you run the command.
+Therefore, before you run the command, you must already have service manifests, config maps, and workload manifests such as `Pod`, `Deployment`, `ReplicaSet`, `Job`, `DaemonSet`, and `StatefulSet` as YAML files in the specified directory.
+. Verify that you can apply these YAML files as-is using the `kubectl apply -f` command. The build-time network policy generator does not work with files that use Helm-style templating.
+. Verify that the service network addresses are not hardcoded. Every workload that needs to connect to a service must specify the service network address as a variable. You can specify this variable by using the workload's resource environment variable or in a config map.
+
+** link:https://github.com/np-guard/cluster-topology-analyzer/blob/main/tests/k8s_guestbook/frontend-deployment.yaml#L25:L28[Example 1: using an environment variable]
```

Comment:
It'd be better to use a permalink to github sources, they might change over time so you might link to stale things here.

Reply:
@kcarmichael08 Is there a way we can work this out?

Reply:
@dhaus67 I'm not familiar with github links - can you let us know what a permalink would be for this example?

Reply:
Here's the perma link for it: https://github.com/np-guard/cluster-topology-analyzer/blob/21678ea900fdda7255914671d9cb359fafe38b6c/tests/k8s_guestbook/frontend-deployment.yaml#L25:L28

FWIW here's the doc on perma links for github: https://docs.github.com/en/repositories/working-with-files/using-files/getting-permanent-links-to-files